src/lib/es2022.sharedmemory.d.ts

```diff
@@ -0,0 +1,86 @@
+interface Atomics {
+    /**
+     * Adds a value to the value at the given position in the array, returning the original value.
+     * Until this atomic operation completes, any other read or write operation against the array
+     * will block.
+     */
+    add(typedArray: BigInt64Array | BigUint64Array, index: number, value: bigint): bigint;

```

Comment:
instead of duplicating these declarations, you should `/// <reference lib="es2020.sharedmemory" />`

Reply:
```ts
interface Atomics {
    /**
     * A non-blocking, asynchronous version of wait and usable on the main thread.
     * Waits asynchronously on a shared memory location and returns a Promise
     * @see wait
     */
    waitAsync(typedArray: BigInt64Array | Int32Array, index: number, value: bigint, timeout?: number): { async: false, value: "ok" | "not-equal" | "timed-out" } | { async: true, value: Promise<"ok" | "not-equal" | "timed-out"> };
}

```

Like the code above ? I looked how they added `` at `` array method in 2022

Reply:
I'm not sure whether you need to reference es2020.sharedmemory. You need to add a test that

1. Specifies `// @lib: es2022.sharedmemory`
2. Uses both Atomics.waitAsync and Atomics.wait to make sure both are available.

Look at $ts/tests/cases/conformance/generators/generatorReturnTypeFallback.1.ts for an example of a test that does (1).
You'll probably want to put the test in tests/cases/conformance/es2022.

Reply:
Added test like the code below: 
```ts
// @target: esnext
// @lib: es2022.sharedmemory
// @noemit: true
// @strict: true

const sab = new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT * 1024);
const int32 = new Int32Array(sab);
Atomics.wait(int32, 0, 0);
await Atomics.waitAsync(int32, 0, 0);
```

I'm not sure what does ``// @noemit: true`` mean/do.
 I've looked some other examples  like ``tests/cases/conformance/es2020/constructBigint.ts`` and ``tests/cases/conformance/generators/generatorReturnTypeInference.ts`` and none of them had this decorator.

Reply:
Also for some reason the CI started breaking 😢  perhaps I did something wrong in ``src/lib/es2022.d.ts``?